text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,359 @@
+# Environment Variables as Context and Baggage Carriers
+
+Add requirements for inter-process context and baggage propagation using
+environment variables as the carrier.
```

Comment:
the purpose of the OTEP is not clear to me. What does "add requirements" mean? Requirements for whom? Is it proposing to extend specification? Why does it need to be extended? 

Context propagation in general is meant to enhance inter-process communication (IPC) with the ability to pass context/baggage. It puts very high-level requirements on the communication protocol, namely to be able to carry additional k-v metadata that propagators use to encode baggage. HTTP as protocol satisfies that requirement, so does parent-child IPC via env vars.

Really the only new thing needed to enable env var IPC is a "carrier" that can deal with case (in)sensitivity of the names, similar to HTTP carrier that does the same. It would be much clearer if the OTEP was explicit about that and limited to just that.

Reply:
@yurishkuro  - thank you for reviewing! The conversation getting started was exactly what I was looking for when copying this forward into a new pull request. After reading your comments & since the original content has now had one review, I've re-written in based on additional context and your feedback. I've also changed the title to make more clear what is being requested. Pretty much everything has been re-written so please feel free to give another review and let me know if this is more clear & and in-line with what you're looking for in an OTEP.